tests/run-pass/concurrency/yield_schedule.rs

```diff
@@ -191,7 +195,10 @@ fn spin_loop() {
             // Note: the thread yield or spin loop hint
             // is required for termination, otherwise
             // this will run forever.
-            std::sync::atomic::spin_loop_hint();
+
+            // Note: should change to use this once cfg(miri) to use intrinsic.
+            //std::sync::atomic::spin_loop_hint();
+            unsafe { miri_yield_thread(); }
```

Comment:
What about using `std::thread::yield_now`?

Reply:
It would work but I wold ideally like to replace this with spin_loop_hint once it works correctly.

Reply:
Sure, but then we can avoid adding `miri_yield_thread` in the first place.^^